js/src/pages/edit-paid-ads-campaign/edit-paid-ads-campaign-form.js

```diff
@@ -82,6 +82,7 @@ const EditPaidAdsCampaignForm = ( props ) => {
 						/>
 						<EditPaidAdsCampaignFormContent
 							formProps={ formProps }
+							allowMultiple={ campaign.allowMultiple }
```

Comment:
💅  ❔ Instead of passing `allowMultiple` / `multiple` down the component tree, what do you think if we put the `campaign.multiple` into form value, and the child components can just pull out `multiple` value from formProps? e.g.:

```
		<Form
			initialValues={ {
				id: campaign.id,
				amount: campaign.amount,
				country: [ campaign.country ],
				multiple: campaign.multiple // put `multiple` into form value.
			} }

Reply:
Not sure if it's a good idea.

Since the coexisting of "old" and "new" campaigns is a special logic. I planned to put all relevant logic at two places, 1) here the `allowMultiple` prop and 2) adding `allowMultiple` property to each campaign entity when receiving API data in actions.js.

That's said, this special logic is isolated from other places. And the `multiple` is designed as a normal prop for `<AudienceSection>` to indicate the selector type, but it doesn't need to know the special logic.

Another thought is, the purpose of `initialValues` seems to be more towards defining form data, and if passing down other props directly is not an obvious bad smell code, I'd suggest we should try to avoid passing prop along with form `initialValues`.

Reply:
> adding `allowMultiple` property to each campaign entity when receiving API data in actions.js.

Ahh, thanks for this, I didn't know that. I thought `allowMultiple` is part of the campaign object returned from the API.

Just want to throw out an idea that I thought about (and related to the `targeted_locations` comment in https://github.com/woocommerce/google-listings-and-ads/pull/1268#issuecomment-1049508806). Since the form is storing the campaign values, we could probably extract the formProps value and determine whether it is a new multi-country campaign or old single-country campaign. Here's a simple illustration of the idea:

```js
// Form.
<Form
	  initialValues={ {
		    id: campaign.id,
		    amount: campaign.amount,
		    country: [ campaign.country ],
		    targeted_locations: campaign.targeted_locations, // NEW: I think we would need this eventually?
	  } }
>...

// Function containing the logic to determine whether it is a new multiple campaign, based on targeted_locations.
const isCampaignMultiple = (campaign) => !! campaign.targeted_locations

// Within child component.
const isMultiple = isCampaignMultiple(formProps.values) // values is essentially campaign here.
```

I'm not sure if the above is a good idea, but I guess it is something that you can evaluate and find out. 😄 

Anyway, with your explanation above, I think we are good with what we have now, we can go with it. 🙂  👍

Reply:
Thanks for the idea! I tried a different way when integrating `targeted_locations` data. Will still use the form value `country`, renamed to `countryCodes`, for showing the single country or multiple countries.

```js
const { amount } = campaign;
const countryCodes = campaign.allowMultiple
	? campaign.targeted_locations
	: [ campaign.country ];

return (
	<Form
		initialValues={ { amount, countryCodes } }
		validate={ handleValidate }
		onSubmit={ handleSubmit }
	>
	// ... omitted
```

Sorry I should have added the relevant design in my head to the context of this PR. 🙏  This approach will be more clearly outlined in the subsequent PR

Reply:
No worry, and thanks for the explanation, we will work through it in the subsequent PR then. 🙂